x-pack/plugin/esql/qa/testFixtures/src/main/resources/stats_percentile.csv-spec

```diff
@@ -1,12 +1,74 @@
 percentileOfNull
-row x = null::integer, y = null, z = 123 - null | stats percIntNull = percentile(y, 90), percNull = percentile(y, 90), percentile(null, 90), percentile(null+2, 90), percentile(z, 90);
+required_capability: extend_aggs_on_constants_support
+row x = null::integer, y = null, z = 123 - null 
+| stats percIntNull = percentile(y, 90), 
+        percNull = percentile(y, 90), 
+        percentile(null, 90), 
+        percentile(null+2, 90), 
+        percentile(z, 90);
 
   percIntNull:double|percNull:double|percentile(null, 90):double|percentile(null+2, 90):double|percentile(z, 90):double
 null                |null           |null                       |null                         |null
 ;
 
+percentileWithPropagateablePercentileArgument
+required_capability: extend_aggs_on_constants_support
+row value200 = 200, x = null::integer, y = null, z = 123 - null, p90 = 80+10 
+| eval p100 = p90 + 10, 
+       p3_null = null + 2, 
+       p50 = 40+10 
+| stats percentile(value200, p90), 
+        percentile(value200, p100), 
+        percentile(value200, p3_null), 
```

Comment:
Shouldn't `percentile(x, null)` be invalid? Similarly for `count_distinct(x, null)`.

Reply:
That's a good question. Here's the paradox: `percentile` value (the second argument) must be a number between 0 and 100. If it's 1000 we are issuing a warning and treat the result of the agg as `null`; by the principle that we issue warnings and return `null`s when things "don't work" I think we should treat `percentile` as `null` if the second parameter is `null`.

Yes, it's not a value between 0 and 100, it's not a missing value, it's.... "unknown" value, a `null`. Which is a bit special than missing or a number between 0 and 100. In `main` this query fails with an ugly exception.

There are other problems or, better said, inconsistencies: `weighted_avg` specifically forbids `null` at data type validation time (`weighted_avg(x, null)`), but it doesn't do it when the value received is the result of an evaluation (`eval n = null + 2 | stats weighted_avg(x, n)`. If `weighted_avg` does check for `null` and forbids it explicitly, `percentile` doesn't even do it.

Reply:
I don't have an answer for you to this question right now. There should be a generic approach to dealing with aggregations' arguments that accept only constants: `top`, `percentile`, `weighted_avg`.

Reply:
The behavior is inconsistent and needs to be fixed, though not in this PR.
If the arguments have to be literals, then throwing an exception (invalid query) would be the way to go.
If not, then folding+ warning is the way to go.

For the former, the function implementation would have to be improved to rely on Validatable interface. Another way, would be marking the field as foldable/not-null and doing this in the planner as it will work across all functions.

Reply:
We don't have to hash this out now, but _maybe_ it'd be safer to start with a validation exception now - we can still go back and return null later, while the other way around _could_ be considered a breaking change, albeit in a very edge case scenario.